watertap/unit_models/mvc/components/complete_condenser.py

```diff
@@ -253,6 +255,19 @@ def initialize(
         )
         init_log.info_high("Initialization Step 1 Complete.")
         # # ---------------------------------------------------------------------
+        # check degrees of freedom
+        under_constrained_flag = False
+        if degrees_of_freedom(blk) != 0:
+            if heat != None:
+                blk.control_volume.heat.fix(heat)
+                under_constrained_flag = True
+            else:
+                raise RuntimeError(
+                    "The model has {} degrees of freedom rather than 0 for initialization."
+                    " This error suggests that an outlet condition has not been fixed"
+                    " for initialization.".format(degrees_of_freedom(blk))
+                )
```

Comment:
This control flow doesn't make much sense to me.

I think `under_constrained_flag` should be renamed `fixed_heat`, and more critically, if the degrees of freedom is greater than 1, and the caller has provided a value for `heat`, we will continue initializing the model.

It seems we need to raise and error if the degrees of freedom is greater than 1, and if it's exactly 1 then we can do this check for heat.

Reply:
I changed the logic and the name.